src/sync.cpp

```diff
@@ -411,16 +414,54 @@ LocalNode* Sync::localnodebypath(LocalNode* l, string* localpath, LocalNode** pa
     }
 }
 
+bool Sync::isscannable(const string& localpath)
+{
+    return localpath.size() < localdebris.size()
+         || memcmp(localpath.data(), localdebris.data(), localdebris.size())
+         || (localpath.size() != localdebris.size()
+          && memcmp(localpath.data() + localdebris.size(),
+                    client->fsaccess->localseparator.data(),
+                    client->fsaccess->localseparator.size()));
+}
+
+void Sync::invalidatefsids()
+{
+    invalidatefsids(localroot);
+}
+
+bool Sync::assignfsids()
+{
+    std::multiset<FileFingerprint*, FileFingerprintCmp> nodes;
+    collectfilenodes(nodes, localroot);
+
+    auto fa = std::unique_ptr<FileAccess>{client->fsaccess->newfileaccess()};
+    bool success;
+    auto rootpath = localroot.localname;
+    if ((success = fa->fopen(&rootpath, true, false)))
+    {
+        if ((success = fa->type == FOLDERNODE))
+        {
+            assignfsids(success, rootpath, std::move(fa), nodes);
+        }
+        else
+        {
+            LOG_err << "rootpath not a folder";
```

Comment:
I think syncs can also be to the whole cloud folder, or an incoming node, not just foldernode?

Reply:
We're only recursing through local folders here so the root must be a foldernode. It's equivalent to the general scanning we do when adding a sync (see Sync::scan). Does that make sense?

Reply:
I think if you set up a sync to your whole cloud drive, the root cloud node type would be ROOTNODE.

Reply:
We're dealing with local nodes here.